engine/lite/request-response/runtime/src/main/scala/pl/touk/nussknacker/engine/requestresponse/ScenarioRoute.scala

```diff
@@ -15,14 +14,35 @@ import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
 import pl.touk.nussknacker.engine.api.process.ProcessName
 import pl.touk.nussknacker.engine.requestresponse.openapi.RequestResponseOpenApiGenerator
 
-import java.nio.file.Files
-
-class ScenarioRoute(handler: RequestResponseAkkaHttpHandler, definitionConfig: OpenApiDefinitionConfig, scenarioName: ProcessName) extends Directives with LazyLogging {
+class ScenarioRoute(handler: RequestResponseAkkaHttpHandler,
+                    config: RequestResponseConfig,
+                    scenarioName: ProcessName)
+  extends Directives with LazyLogging {
 
   // Regarding https://spec.openapis.org/oas/v3.1.0#serverObject server url accept urls "relative to the location where the OpenAPI document is being served"
   // We use this relative reference instead of default '/' because runtime can by server reverse proxies (i.e. ingress controller) that can rewrite this url
   private val defaultServerUrl = "./"
 
+  val securityDirectiveOpt: Option[AuthenticationDirective[String]] = {
+
+    val password = config.basicAuthConfig.map(_.password)
+    val user = config.basicAuthConfig.map(_.user)
+
+    def rrAuthenticator(credentials: Credentials): Option[String] =
```

Comment:
this _.get doesn't look too nice... I know they won't fail, but I looks a bit dangerous. 
Let's extract method:
```
prepareAuthenticator(basicAuthConfig): Credentials => Option[String]
```
and invoke it in line 39 (and then you won't have `flatMap { _ ` which also looks a bit suspicious

Reply:
refactored